### PR TITLE
Upped the Go version and included TLS certificate support

### PR DESCRIPTION
**Pull request by JohnSharpe**
Added TLS and re-added SSL testing with a self-signed, 980-year certificate.

### CODE REVIEW

File: provider/resource_rediscloud_pro_database.go [hunk 1]
```diff
@@ -770,18 +779,28 @@ func resourceRedisCloudProDatabaseUpdate(ctx context.Context, d *schema.Resource
...
 	if enableTLS {
 		// TLS only: enable_tls=true, client_ssl_certificate="".
 		update.EnableTls = redis.Bool(enableTLS)
-		// mTLS: enableTls=true, non-empty client_ssl_certificate.
+
 		if clientSSLCertificate != "" {
+			// mTLS: enableTls=true, non-empty client_tls_certificates.
 			update.ClientSSLCertificate = redis.String(clientSSLCertificate)
+
+			// TODO If the user has enabled TLS and provided an SSL certificate, we might want to scrub any TLS certificates
```
Comment (greg-oc) [inline provider/resource_rediscloud_pro_database.go:792]: Should this still be here?

File: provider/resource_rediscloud_pro_tls_test.go [hunk 2]
```diff
@@ -15,56 +14,41 @@ import (
...
...vmdRRZWZaiDm5Gx45aNRpcz9UHw0kjkG2HPW+PiZoaeQ\\nRcUTLOfr+Z7SDIBlyBNwzNt8j6s88SDTin9tp5oNL/WnCtNLkjf9SjN+nj3nxopW\\n9s7ocjV3nbBwfODI/t0u5yVwmM/xvrI1lail7IXwqHV2v1DTnh5BELstw+8i9NI0\\nj1dhIVQhwzsu5tgwQig8iXQTg6/kVxNnMgcUQdEJckk3aonjECOCJjs9aDULAbA1\\n9mcZnx57WeugJj8eMBeixIoRzHJcYTx92Mcrr0hUHi+OmIxTpu7ydRMCAwEAAaNT\\nMFEwHQYDVR0OBBYEFNA3DgT/I4yju8Im0fKb9u08WiPUMB8GA1UdIwQYMBaAFNA3\\nDgT/I4yju8Im0fKb9u08WiPUMA8GA1UdEwEB/wQFMAMBAf8wDQYJKoZIhvcNAQEL\\nBQADggIBAIIwf2fKxLpgh7kTmMRI7zHPlzwEMo2xzgjT1JifpMTHjv+xusdrYdRZ\\nN92vLy3BZ89XcdlVuYOQYdrrniWZ9hDk/lWP5PyAvyFkVAMijmwcySW5BioddDfc\\n7augJmaP2X8Qg0CwOAJazC9RSV6x31G9ah5x89Nsh0uQc5e4udLQsV68DOD9S7Go\\nQxFB8qK/Xx8Z0OytCch9Oh/yZZzL5xBZtla5TWFG8kgoaj7m91lddhX+px04l/fC\\n48zxHRMJSjr5O8SUX3AKx189D9aZEXWLVyfdDtJ7yJmbhOVMjB60+20Jqa1fgb0h\\n1Hh6E+TP5ObDFni3ocjcmnSwwBr9Ih6PlES/z77AK4KiA7S0A+MZQVshLN6n9GVw\\nK78HS19IAHO0A8BKdxphaGBKJzye8+/S6Meyemq2hysHczNFeYWU13UKu9daWOYS\\nPlmhjikHCOii7eipK0+GtTfmkgYmL6f7OykkZ+pVjYtiq7qTU5ZIlWlW2uoPh2Oq\\ngVf//6zduKBxxEcl0i0qDHclx144uCnDnibhlnXcngqexMqNZWEn2Ld/7/mm+jYN\\nMnA37eTHAJrJ+urvEmkdonF5FFUpZtet53abyd0eYzRrVXof6iroQcetgnJA+k+I\\n4HrYoxJnDrzHJ+ycJ457/tggup254bgeqmCzalLTUeVNr9H2/lbT\\n-----END CERTIFICATE-----"
```
Comment (greg-oc) [inline provider/resource_rediscloud_pro_tls_test.go:17]: just double checking this is not an actual valid certificate?

File: provider/resource_rediscloud_pro_tls_test.go [hunk 3]
```diff
@@ -224,34 +197,149 @@ func TestAccResourceRedisCloudSubscription_createWithDatabaseWithEnabledTlsAndIn
...
+		Steps: []resource.TestStep{
+			{
+				Config:      fmt.Sprintf(testAccResourceRedisCloudSubscriptionOneDbWithoutEnableTlsAndWithCert, testCloudAccountName, name, 1, password, invalidSslCertificate),
+				ExpectError: regexp.MustCompile("Error: 400 BAD_REQUEST - DATABASE_INVALID_CERT: Database certificate is invalid"),
+			},
+		},
+	})
+}
+
+// enable_tls=false, client_ssl_certificate="", client_tls_certificates=["something"]
```
Comment (greg-oc) [inline provider/resource_rediscloud_pro_tls_test.go:220]: should this still be here?

File: provider/resource_rediscloud_pro_tls_test.go [hunk 4]
```diff
@@ -224,34 +197,149 @@ func TestAccResourceRedisCloudSubscription_createWithDatabaseWithEnabledTlsAndIn
...
-				ExpectError: regexp.MustCompile("Error: 400 BAD_REQUEST - DATABASE_INVALID_CERT: Database certificate is invalid"),
+				Config:      fmt.Sprintf(bothSslAndTls, testCloudAccountName, name, 1, password, sslCertificate, sslCertificate),
+				ExpectError: regexp.MustCompile("Conflicting configuration arguments"),
 			},
 		},
 	})
 }
 
-const testAccResourceRedisCloudSubscriptionOneDbWithEnableTlsAndCert = `
+// enable_tls=true, client_ssl_certificate="", client_tls_certificates=["something"]
```
Comment (greg-oc) [inline provider/resource_rediscloud_pro_tls_test.go:265]: should this still be here?

**Review by greg-oc — APPROVED** [review]
A few small questions but looks good

File: provider/resource_rediscloud_pro_tls_test.go (hunk 2 above)
Comment (JohnSharpe) [reply provider/resource_rediscloud_pro_tls_test.go:17]: Hahaha no, just a self-signed, 980-year-valid cert I generated

File: provider/resource_rediscloud_pro_tls_test.go (hunk 3 above)
Comment (JohnSharpe) [reply provider/resource_rediscloud_pro_tls_test.go:220]: Yeah, this just shows the combination of relevant properties being attempted

File: provider/resource_rediscloud_pro_database.go (hunk 1 above)
Comment (JohnSharpe) [reply provider/resource_rediscloud_pro_database.go:792]: Nope, good catch, cheers!

File: provider/resource_rediscloud_pro_tls_test.go (hunk 4 above)
Comment (JohnSharpe) [reply provider/resource_rediscloud_pro_tls_test.go:265]: Same as above

**Pull request merged**